src/native-common/Image.tsx

```diff
@@ -202,7 +202,16 @@ export class Image extends React.Component<Types.ImageProps, ImageState> impleme
 
         const source: RN.ImageSourcePropType = { uri: this.props.source };
         if (this.props.headers) {
-            source.headers = this.props.headers;
+            const cacheControlHeader = this._getMaxStaleHeader();
+            if (cacheControlHeader) {
+                // Filter out Cache-Control: max-stale. It has the opposite effect on iOS: instead of having
```

Comment:
Apologies for not catching this when working on #853 
Basically it looks like NSURLRequest disables local cache if it sees a caching header it doesn't understand. So not only doesn't Cache-Control: max-stale do what it's spec'ed to, it actually hurts and has to be removed. cc @berickson1